src/topology/inseparable.lean

```diff
@@ -178,12 +179,199 @@ def specialization_preorder : preorder X :=
 
 variable {X}
 
+local attribute [instance] specialization_preorder
+
 /-- A continuous function is monotone with respect to the specialization preorders on the domain and
 the codomain. -/
-lemma continuous.specialization_monotone (hf : continuous f) :
-  @monotone _ _ (specialization_preorder X) (specialization_preorder Y) f :=
+lemma continuous.specialization_monotone (hf : continuous f) : monotone f :=
 λ x y h, h.map hf
 
+lemma closure_singleton_eq_Iic (x : X) : closure {x} = Iic x :=
+ext $ λ _, specializes_iff_mem_closure.symm
+
+/-- A subset `S` of a topological space is stable under specialization
+if `x ∈ S → y ∈ S` for all `x ⤳ y`. -/
+def stable_under_specialization (s : set X) : Prop :=
+∀ ⦃x y⦄, x ⤳ y → x ∈ s → y ∈ s
+
+/-- A subset `S` of a topological space is stable under specialization
+if `x ∈ S → y ∈ S` for all `y ⤳ x`. -/
+def stable_under_generalization (s : set X) : Prop :=
+∀ ⦃x y⦄, y ⤳ x → x ∈ s → y ∈ s
+
+lemma stable_under_specialization_iff_is_lower_set {s : set X} :
+  stable_under_specialization s ↔ is_lower_set s :=
+iff.rfl
+
+lemma stable_under_generalization_iff_is_upper_set {s : set X} :
+  stable_under_generalization s ↔ is_upper_set s :=
+iff.rfl
```

Comment:
This lemma looks wrong. Either `specialization_order` is made a global instance (bad), or no lemma is stated using it. In the latter case, you can instead state lemmas using a type synonym equipped with the specialisation order. See `is_upper_set_set_of` for inspiration.

Reply:
This is intended to be used when `specialization_order` is made a local instance.

Reply:
Made them into examples instead.